scarlet/lite/frame.py

```diff
@@ -0,0 +1,258 @@
+            Whether to calculate the gradient of the radius**2 (``useR2==True``)
+            or the radius (``useR2==False``).
+
+        Returns
+        -------
+        result: float
+            The gradient of the likelihood wrt the semi-major axis.
+        """
+        grad = - 2 / self._major * self._xa ** 2
+        if use_r2:
+            grad *= 2
+        else:
+            r = self.R
+            grad *= 1 / r
+        return np.sum(grad * input_grad)
+
+    def grad_minor(self, input_grad, use_r2):
+        """The gradient of either the radius or radius**2 wrt. the semi-minor axis
+
+        Parameters
+        ----------
+        input_grad: np.array
+            Gradient of the likelihood wrt the component model
+        use_r2: bool
+            Whether to calculate the gradient of the radius**2 (``useR2==True``)
+            or the radius (``useR2==False``).
+
+        Returns
+        -------
+        result: float
+            The gradient of the likelihood wrt the semi-minor axis.
+        """
+        grad = -2 / self._minor * self._yb ** 2
+        if use_r2:
+            grad *= 2
+        else:
+            r = self.R
+            grad *= 1 / r
+        return np.sum(grad * input_grad)
+
+    def grad_theta(self, input_grad, use_r2):
+        """The gradient of either the radius or radius**2 wrt. the rotation angle
+
+        Parameters
+        ----------
+        input_grad: np.array
+            Gradient of the likelihood wrt the component model
+        use_r2: bool
+            Whether to calculate the gradient of the radius**2 (``useR2==True``)
+            or the radius (``useR2==False``).
+
+        Returns
+        -------
+        result: float
+            The gradient of the likelihood wrt the rotation angle.
+        """
+        grad = self._xa * self._xMinor / self._major - self._yb * self._xMajor / self._minor
+        if use_r2:
+            grad *= 2
+        else:
+            r = self.R
+            grad *= 1 / r
+        return np.sum(grad * input_grad)
+
+    @property
+    def x0(self):
+        """The x-center"""
+        return self._x0
+
+    @property
+    def y0(self):
+        """The y-center"""
+        return self._y0
+
+    @property
+    def major(self):
+        """The semi-major axis"""
+        return self._major
+
+    @property
+    def minor(self):
+        """The semi-minor axis"""
+        return self._minor
+
+    @property
+    def theta(self):
+        """The rotation angle"""
+        return self._theta
+
+    @property
+    def bbox(self):
+        """The bounding box to old the model"""
+        return self._bbox
+
+    @property
+    def R(self):
+        """The radial coordinates of each pixel"""
+        if self._radius is None:
+            self._radius = np.sqrt(self._radius2) + self._rMin
+        return self._radius
+
+    @property
+    def R2(self):
+        """The radius squared located at each pixel"""
+        return self._radius2 + self._rMin
```

Comment:
These two minima aren't consistent. It should be `self._radius2 + self._rMin**2` and just `np.sqrt(self._radius2)` above.

Reply:
I don't think you need added complexity, though? Just store `_rMin2`, have `R2` return `self._radius2 + self._rMin2`, and `R` return `np.sqrt(self._radius2 + self._rMin2)` (or `np.sqrt(self.R2)`).

Reply:
I should have read this comment better, you're right and I fixed it.